power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/services/UserRegistrationServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+package com.adorsys.gis.powerpay.powerpaybackend.services;
+
+import java.security.SecureRandom;
+
+import org.springframework.stereotype.Service;
+
+import com.adorsys.gis.powerpay.powerpaybackend.domain.User;
+import com.adorsys.gis.powerpay.powerpaybackend.domain.UserRegistration;
+import com.adorsys.gis.powerpay.powerpaybackend.repository.UserRegistrationRepository;
+import com.adorsys.gis.powerpay.powerpaybackend.repository.UserRepository;
+import com.adorsys.gis.powerpay.powerpaybackend.utils.DataSecurityService;
+import com.adorsys.gis.powerpay.powerpaybackend.utils.UserRegistrationException;
+
+import jakarta.validation.constraints.NotNull;
+import lombok.AllArgsConstructor;
+
+@Service
+@AllArgsConstructor
+public class UserRegistrationServiceImpl implements UserRegistrationService {
+
+    private final UserRegistrationRepository userRegistrationRepository;
+    private final UserRepository userRepository;
+    private final DataSecurityService dataSecurityService;
+
+    private static final String CHARACTERS = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+    private static final int OTP_LENGTH = 8;
+
+
+    @Override
+    public User registerUser(@NotNull String registrationId, String pin, String otp) {
+
+        User newUser = new User();
+        UserRegistration userRegistration = new UserRegistration();
+
+        try {
+            userRepository.save(newUser);
+        } catch (Exception e) {
+            throw new UserRegistrationException("User registration failed", e);
+        }
+
+        return newUser;
+    }
```

Comment:
This logic is missing a lot of points. 
1. You should add attributes to the created user object (and hash the pin while you're at it)
2. You should not create a userRegistration object here, but call the related repo to find the one corresponding to the provided ID.